tests/dialects/test_builtin.py

```diff
@@ -167,6 +167,14 @@ def test_IntegerAttr_normalize():
         IntegerAttr(256, 8)
 
 
+def test_IntegerAttr_to_bool():
+    assert IntegerAttr.from_bool(False).to_bool() is False
+    assert IntegerAttr.from_bool(True).to_bool() is True
+
+    with pytest.raises(AssertionError):
+        IntegerAttr(1, i8).to_bool()
+
```

Comment:
Yeah I don't think we need to be this strict, I'm happy to follow C/Python's truthiness as being non-zero.